lib/galaxy/jobs/__init__.py

```diff
@@ -469,13 +469,20 @@ def __get_params(self, parent):
         rval = {}
         for param in parent.findall('param'):
             key = param.get('id')
-            param_value = param.text
+            if key in ["container", "container_override"]:
+                from galaxy.tools.deps import requirements
+                containers = map(requirements.container_from_element, list(param))
```

Comment:
@jmchilton I was looking at this code because it's not Python3-compatible, and it seems wrong to me: shouldn't it work on the `<container>` elements inside `<param id="container">` ?

Reply:
I think it does do that - I added a comment to clarify this and fixed the iterator vs. list problem in https://github.com/galaxyproject/galaxy/pull/5427. I could go a step further and filter the inner elements on the tag name to ensure they are container tags but right now I'm just assuming that is what they are because I don't think any other tag is valid in there.